### ISSUE TITLE
Syntax error on -copy

### ISSUE THREAD

**Issue by charlieMonroe** [drkameleon/iswift#112]
Example:

```
NSArray *arr1 = @[ ];
NSArray *arr2 = [arr1 copy];
```

In this particular example (`NSArray`, `NSDictionary`, `NSString`), it's OK to leave out the `-copy` call:

```
let arr1 = [ ]
let arr2 = arr1
```

since these structures are actually `struct`s in Swift and they get copied when re-assigned to a new variable.

Unfortunately, for the rest, this isn't as easy, since in some cases this may make a difference - in particular, when you are using your own classes, make a copy and then modify variables.

So, in the best scenario:

- leave out `-copy` for `NSArray`, `NSDictionary` and `NSString`
- keep it as a method call for the rest

**Comment by drkameleon** [comment]
Hmmm... This is rather tricky.

For example, here:
```objective-c
NSArray *arr2 = [arr1 copy];
```

How do we know `arr1`'s type? The compiler does. iSwift - being a parser - doesn't. (And it would be rather complicated to start making a list with all user-defined known variable types...)

**Comment by charlieMonroe** [comment]
Yeah, that's a problem, you don't know that... :(

**Comment by drkameleon** [comment]
Issue (`copy` throwing an error as a message selector) fixed as of the upcoming 2.0.

---- 

P.S. For the rest of it, let's say I'll see it in the future... ;)
